includes/class-newspack-popups-model.php

```diff
@@ -12,6 +12,87 @@
  */
 final class Newspack_Popups_Model {
 
+	/**
+	 * Retrieve all Popus (first 100).
+	 *
+	 * @return array Array of Popup objects.
+	 */
+	public static function retrieve_popups() {
+		$args = [
+			'post_type'      => Newspack_Popups::NEWSPACK_PLUGINS_CPT,
+			'post_status'    => 'publish',
+			'posts_per_page' => 100,
+		];
+
+		$popups = self::retrieve_popup_with_query( new WP_Query( $args ), true );
+
+		// Hacky check if this is a single popup object as opposed to an array of Pop-ups. TODO: replace.
+		if ( isset( $popups['id'] ) ) {
+			$popups = [ $popups ];
+		}
+		foreach ( $popups as &$popup ) {
+			if ( ! count( $popup['categories'] ) ) {
+				$popup['sitewide_default'] = true;
+				break;
+			}
+		}
+		return $popups;
+	}
+
+	/**
+	 * Set post time to now, making it the sitewide popup.
+	 *
+	 * @param integer $id ID of the Popup to make sitewide default.
+	 */
+	public static function set_sitewide_popup( $id ) {
+		$popup = self::retrieve_popup_by_id( $id );
+		if ( ! $popup ) {
+			return new \WP_Error(
+				'newspack_popups_popup_doesnt_exist',
+				esc_html__( 'The Popup specified does not exist.', 'newspack-popups' ),
+				[
+					'status' => 400,
+					'level'  => 'fatal',
+				]
+			);
+		}
+		$time = current_time( 'mysql' );
+		wp_update_post(
+			[
+				'ID'            => $id,
+				'post_date'     => $time,
+				'post_date_gmt' => get_gmt_from_date( $time ),
+			]
+		);
+	}
```

Comment:
I just realized this sets the sitewide popup by making it the newest popup. I think that is super clever. 

If someone sets the sitewide default and then creates a new popup with no category, the new popup will take over as the sitewide default. Is this the intended behavior? (Both ways makes sense IMO, so just a heads-up)

Reply:
Yes, I think this behavior is OK. It might  be a  little unexpected, but it would be weirder if sitewide default was the newest Pop-up unless you had used the selection UI in the Newspack Plugin. That feels less predictable. Do you think we should iterate on this logic and try to improve it?

Reply:
No let's go with this first and see if anyone gets confused. I think it'll be fine.